cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocConf.java

```diff
@@ -421,8 +424,9 @@ static DataprocConf create(Map<String, String> properties) {
   static DataprocConf create(Map<String, String> properties, @Nullable SSHPublicKey publicKey) {
     String accountKey = getString(properties, "accountKey");
     if (accountKey == null || AUTO_DETECT.equals(accountKey)) {
+      String endPoint = getString(properties, TOKEN_ENDPOINT_KEY);
       try {
-        getComputeEngineCredentials();
+        new ComputeEngineCredentials(endPoint).refreshAccessToken();
       } catch (IOException e) {
         throw new IllegalArgumentException(e.getMessage(), e);
```

Comment:
I would suggest to have a better error message that indicates it failed to get the access token and the service account key was not provided instead of using `e.getMessage()`.